### PR TITLE
[CROSSDATA-162] [Mongo Connector tasks] Support for modifier NOT

### PR DESCRIPTION
**Pull request by pmadrigal**
* Support NOT filter by collect(spark)

### CODE REVIEW

File: mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/MongoQueryProcessor.scala [hunk 1]
```diff
@@ -38,47 +42,66 @@ object MongoQueryProcessor {
...
   def apply(logicalPlan: LogicalPlan, config: Config, schemaProvided: Option[StructType] = None) = new MongoQueryProcessor(logicalPlan, config, schemaProvided)
 
-  def buildNativeQuery(requiredColums: Seq[ColumnName], filters: Array[SourceFilter]): (DBObject, DBObject) =
-    (filtersToDBObject(filters), selectFields(requiredColums))
+  def buildNativeQuery(requiredColums: Seq[ColumnName], filters: Array[SourceFilter], config: Config): (DBObject, DBObject) = {
+    (filtersToDBObject(config, filters), selectFields(requiredColums))
+  }
 
-  private def filtersToDBObject(sFilters: Array[SourceFilter]): DBObject = {
+  def filtersToDBObject(config: Config, sFilters: Array[SourceFilter], parentFilterIsNot: Boolean = false): DBObject = {
```
Comment (luismcl) [inline mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/MongoQueryProcessor.scala]: Could rename the variable name?
negateAllQuery or other name... I don't know...

File: mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/MongoQueryProcessor.scala [hunk 2]
```diff
@@ -38,47 +42,66 @@ object MongoQueryProcessor {
...
 
-  private def convertToStandardType(value: Any): Any = value
+    /**
+      * Check if the field is "_id" and if the user wants to filter by this field as an ObjectId
+      *
+      * @param attribute Name of the file
+      * @param value Value for the attribute
+      * @return The value in the correct data type
+      */
+     private def checkObjectID(attribute: String, value: Any, config: Config) : Any = {
```
Comment (luismcl) [inline mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/MongoQueryProcessor.scala]: Same here, this function has, how Pablo says?, Collateral effects? the name indicates just a verification, but builds a new Object.

**Comment by pfcoperez** [comment]
+1

**Comment by darroyocazorla** [comment]
+1

**Pull request merged**